src/h5web/visualizations/index.test.ts

```diff
@@ -304,6 +320,189 @@ describe('Visualization definitions', () => {
+      const group = makeGroup(
+        'nonValidSpectrum',
+        [makeStrAttr('signal', 'my_signal')], // missing NXdata attribute
+        [makeDatasetHardLink('my_signal', NxSpectrumDataset.id)]
+      );
+
+      const entry = makeNxEntry('entry', {
+        defaultPath: '/entry/image',
+        ids: { nonValidSpectrum: group.id },
+      });
+
+      const root = makeNxRoot('root', {
+        ids: { entry: entry.id },
+      });
+
+      const metadata = makeMetadata({
+        root: root.id,
+        datasets: [NxSpectrumDataset],
+        groups: [root, group, entry],
+      });
+
+      expect(supportsEntity(entry, metadata)).toBe(false);
+    });
+
+    it('should support a NXroot with a default attribute pointing to a valid NXentity with a valid default (relative paths)', () => {
+      const group = makeNxDataGroup('spectrum', {
+        signal: 'my_signal',
+        ids: { my_signal: NxSpectrumDataset.id },
+      });
+
+      const intermediateGroup = makeGroup(
+        'intermediate',
+        [],
+        [makeGroupHardLink('spectrum', group.id)]
+      );
+
+      const entry = makeNxEntry('entry', {
+        defaultPath: 'intermediate/spectrum',
+        ids: { intermediate: intermediateGroup.id },
+      });
+
+      const root = makeNxRoot('root', {
+        defaultPath: 'entry',
+        ids: { entry: entry.id },
+      });
+
+      const metadata = makeMetadata({
+        root: root.id,
+        datasets: [NxSpectrumDataset],
+        groups: [root, entry, intermediateGroup, group],
+      });
+
+      expect(supportsEntity(root, metadata)).toBe(true);
+    });
+
+    it('should support a NXroot with a default attribute pointing to a valid NXentity with a valid default (absolute paths)', () => {
+      const group = makeNxDataGroup('spectrum', {
+        signal: 'my_signal',
+        ids: { my_signal: NxSpectrumDataset.id },
+      });
+
+      const intermediateGroup = makeGroup(
+        'intermediate',
+        [],
+        [makeGroupHardLink('spectrum', group.id)]
+      );
+
+      const entry = makeNxEntry('entry', {
+        defaultPath: '/entry/intermediate/spectrum',
+        ids: { intermediate: intermediateGroup.id },
+      });
+
+      const root = makeNxRoot('root', {
+        defaultPath: '/entry',
+        ids: { entry: entry.id },
+      });
+
+      const metadata = makeMetadata({
+        root: root.id,
+        datasets: [NxSpectrumDataset],
+        groups: [root, entry, intermediateGroup, group],
+      });
+
+      expect(supportsEntity(root, metadata)).toBe(true);
+    });
+
+    it('should not support a NXroot without default attribute', () => {
+      const group = makeNxDataGroup('spectrum', {
+        signal: 'my_signal',
+        ids: { my_signal: NxSpectrumDataset.id },
+      });
+
+      const intermediateGroup = makeGroup(
+        'intermediate',
+        [],
+        [makeGroupHardLink('spectrum', group.id)]
+      );
```

Comment:
Perhaps the intermediate group is not needed for this test.

Reply:
True. I made a dedicated tests for "deep" paths.